connection.go

```diff
@@ -740,16 +740,14 @@ func (s *Connection) shutdown(closeTimeout time.Duration) {
 
 	if err != nil {
 		duration := 10 * time.Minute
-		time.AfterFunc(duration, func() {
```

Comment:
this avoids that if `s.shutdownChan <- err` is drained, we don't have a goroutine leaked during 10 minutes

https://go.dev/play/p/3PcdQYbcZdu

Reply:
✅